hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -105,7 +105,9 @@ class DefaultSource extends RelationProvider
     val tableType = metaClient.getTableType
     val queryType = parameters(QUERY_TYPE_OPT_KEY)
     log.info(s"Is bootstrapped table => $isBootstrappedTable, tableType is: $tableType")
-
+    val schemaUtil = new TableSchemaResolver(metaClient)
+    schemaUtil.getTableAvroSchema(false) // this will throw InValidTableException if there is no
```

Comment:
Verifying table existence via schema resolution, seems pretty un-intuitive to me. can we do it from the `metaClient`?